Sources/ProtobufferCoding/Decoder/KeyedProtoDecodingContainer.swift

```diff
@@ -93,7 +93,15 @@ class KeyedProtoDecodingContainer<Key: CodingKey>: InternalProtoDecodingContaine
     }
 
     func decode(_ type: Int.Type, forKey key: Key) throws -> Int {
-        throw ProtoError.decodingError("Int not supported, use Int32 or Int64")
+        let keyValue = try extractIntValue(from: key)
+        if let value = data[keyValue]?.last {
+            if MemoryLayout<Int>.size == 4 {
+                return try Int(decodeInt32(value))
+            } else if MemoryLayout<Int>.size == 8 {
+                return try Int(decodeInt64(value))
+            }
+        }
+        throw ProtoError.decodingError("No data for given key")
```

Comment:
This code snipped is used quite often as part of the PR. If possible, I would suggest to move it to a throwable initializer of the type `Int` and reuse that initializer throughout the PR? The same applies to the same `if else` constructs that are used throughout the PR for other `Int`ish types.

Reply:
I moved the width-dependent decoding into extensions of the respective types, and the encoding parts into the shared internal en-/decoder containers. This reduces the code duplication across the several coding containers.